### DISCUSSION TITLE
How can I give image to monai transforms?

### DISCUSSION THREAD

**Discussion by JuAnYoO** [Project-MONAI/MONAI#6810] [Q&A] [answered]
test_transforms = Compose(
    [
        LoadImaged(keys=["image1","image2"]),
        EnsureChannelFirstd(keys=["image1","image2"]),
        EnsureTyped(keys=["image1","image2"]),
        ScaleIntensityd(keys=["image1","image2"],minv=0.0, maxv=1.0),
        Resized(
            keys=["image1","image2"],
            spatial_size=(512,512),

        ),
        CenterSpatialCropd(keys=["image1","image2"],
                         roi_size=(448,448), 
                        ),          
        ToTensord(keys=["image1", "image2"]),
    ]
)

def sgmt(r_img, l_img):
    test_Data =[
        {
            "image1": r_img,
            "image2": l_img    
        }
    ]
    img = test_transforms(test_Data[0])

This is my code and r_img and l_img are png images. 
When I run this code, "TypeError: expected str, bytes or os.PathLike object, not numpy.ndarray" error occurs. How can I directly give image to transforms?

**Comment by KumoLiu** [comment]
Hi @JuAnYoO, could you please remove `LoadImaged` in the `test_transforms` and try again?
Thanks!

**Reply by JuAnYoO** [reply]
ValueError: Metadata not available and channel_dim=None, EnsureChannelFirst is not in use.
This error occured

**Reply by KumoLiu** [reply, marked as answer]
You could specify `channel_dim` in `EnsureChannelFirst `.
https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/monai/transforms/utility/array.py#L184-L187
